WooCommerce/WooCommerceTests/ViewRelated/Orders/Order Details/OrderDetailsDataSourceTests.swift

```diff
@@ -327,6 +327,20 @@ final class OrderDetailsDataSourceTests: XCTestCase {
         XCTAssertNil(createShippingLabelRow)
     }
 
+    func test_create_shipping_label_button_is_not_visible_for_cash_on_delivery_order() throws {
+        // Given
+        let order = makeOrder().copy(status: .processing, datePaid: .some(nil), paymentMethodID: "cod")
+        let dataSource = OrderDetailsDataSource(order: order, storageManager: storageManager)
+        dataSource.isEligibleForShippingLabelCreation = false
```

Comment:
For this test to best reflect the COD order scenario, I'd suggest testing these conditions:

```
dataSource.isEligibleForShippingLabelCreation = true
dataSource.isEligibleForCardPresentPayment = true
```

That way we're testing that even if the order is otherwise eligible for shipping label creation, the button doesn't appear because it's eligible for card-present payment.

Reply:
Thanks for taking a look @rachelmcr Great catch!

I pushed an update to the test that:
* sets dataSource.isEligibleForShippingLabelCreation to true
* ensures the order is eligible for In-Person Payments by setting its total amount, and by inserting a valid payment gateway into the moved coredata context (dataSource.isEligibleForCardPresentPayment is read-only so it can not be set directly)